### DISCUSSION TITLE
Cluster client support for redis transactions

### DISCUSSION THREAD

**Discussion by kamkash** [NodeRedis/node_redis#2384] [Q&A] [answered]
node-redis experts,

Does node-redis cluster client support Redis transactions?

by redis transactions I mean support for redis commands MULTI/EXEC and WATCH across cluster nodes/shards

**Comment by leibale** [comment, marked as answer]
Redis (the server) does not support cross-slot commands in the same transaction (even if the slots are currently in the same shard/node). If you want to make sure that a couple of keys are in the same slot, you can use ["hash tags"](https://redis.io/docs/reference/cluster-spec/#hash-tags)
```javascript
// this will throw `CROSSSLOT` error
cluster.multi()
  .set('11')
  .set('12')
  .exec();

// this will work
cluster.multi()
  .set('{1}1')
  .set('{1}2')
  .exec();
```

**Reply by kamkash** [reply]
Excellent thank you for the quick answer.

I asked because the Python Redis Cluster client Redis.py does not support MULTI even with "hash tags"  (as far as I know). 
I think the cluster client supports fetching a specific master node client and executing transactions on the specific node.
I was trying to understand if this is an issue with the client or if there is a bigger issue with Redis (the server).

**Reply by chayim** [reply]
redis-py supports pipelining with hashes. You can find the full coverage layout [here](https://github.com/redis/redis-py/blob/18560b43ed759f4ace7ac16f4a3dd53ab2841969/tests/test_cluster.py#L2646) - which shows those items overriden.

**Reply by kamkash** [reply]
@chayim  

Thanks for the info 

I know redis.py cluster supports pipelining. 
I am trying to understand support for transactions and Pipelining in a cluster:  MULTI/EXEC/WATCH  and PIPELINE

I ran into this RedisCluster exception using pipeline with transaction: [here](https://github.com/redis/redis-py/blob/18560b43ed759f4ace7ac16f4a3dd53ab2841969/redis/cluster.py#L746)

Do you know if Redis MULTI/EXEC command is supported in redis.py Clusters?
